### PR TITLE
[SDP-823] Prevent channel account cmd's from being run in parallel

### PR DESCRIPTION
**Pull request by ziyliu**
### What
Have each channel account command grab an advisory lock before any relevant business logic is run when we instantiate the service instance at command setup. 

### Why
This will force any competing parallel processes to error out immediately if an existing cmd process is still running as shown below. This will prevent race conditions and any other issues arising from conflicting commands.

#### process 1 
```
DEBU[2023-08-21T14:22:38.859-07:00] Using default log level: "trace"              pid=27130
INFO[2023-08-21T14:22:38.859-07:00] Version: 1.0.0-rc1                            pid=27130
INFO[2023-08-21T14:22:38.859-07:00] GitCommit:                                    pid=27130
WARN[2023-08-21T14:22:38.864-07:00] Using "DRY_RUN" crash tracker                 pid=27130
INFO[2023-08-21T14:23:38.882-07:00] Found 2 accounts to delete...                 pid=27130
INFO[2023-08-21T14:23:42.019-07:00] Successfully deleted channel account "GC5T375FBFAVKDA7PNF6PUVZ4NYXKJGT5QHLN2D3QXIIBFZJULBDK5YX"  pid=27130
INFO[2023-08-21T14:23:46.744-07:00] Successfully deleted channel account "GCBSQXO3YKE4SBYHXT6Y6ERGEGL2BXFMDOCV4765UVPBT7UELH6FNE6F"  pid=27130
```

#### process 2 (running concurrently)
```
DEBU[2023-08-21T14:22:44.149-07:00] Using default log level: "trace"              pid=27163
INFO[2023-08-21T14:22:44.149-07:00] Version: 1.0.0-rc1                            pid=27163
INFO[2023-08-21T14:22:44.149-07:00] GitCommit:                                    pid=27163
WARN[2023-08-21T14:22:44.156-07:00] Using "DRY_RUN" crash tracker                 pid=27163
ERRO[2023-08-21T14:22:44.157-07:00] Cmd channel-accounts delete crash: failed getting db advisory lock in DeleteChannelAccounts: cannot retrieve unavailable db advisory lock  pid=27163
FATA[2023-08-21T14:22:44.157-07:00] Error deleting channel account: failed getting db advisory lock in DeleteChannelAccounts: cannot retrieve unavailable db advisory lock  pid=27163
exit status 1
```

### Known limitations

[TODO or N/A]

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [ ] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/transactionsubmission/utils/utils.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 	sdpUtils "github.com/stellar/stellar-disbursement-platform-backend/internal/utils"
 )
 
+const AdvisoryLock = int(2172398390434160)
```
Comment (marcelosalloum) [inline internal/transactionsubmission/utils/utils.go:12]: 💡  If this constant is only relevant to the channel accounts service, I believe it should live in the `channel_accounts_service.go` file.

File: internal/transactionsubmission/services/channel_account_service.go
Comment (marcelosalloum) [inline internal/transactionsubmission/services/channel_account_service.go:1]: You added the advisory lock enforcement to the create and verify commands, but didn't add it to the ensure and delete commands, right?

I think we can enforce it for the whole channelAccount service by adding it in a single place: `NewChannelAccountService`.

The `view` doesn't need it, but it runs so fast that it wouldn't have much impact to add it there as well. WDYT?

**Review by marcelosalloum — COMMENTED** [review]
Pretty cool! Thanks for adding that!

I've added a suggestion to acquire the advisory lock in the NewChannelAccountService, so it's applied to the whole service.

File: internal/transactionsubmission/services/channel_account_service.go
Comment (ziyliu) [reply internal/transactionsubmission/services/channel_account_service.go:1]: no, I added it to every command but `view`. but sure, we can also do this when creating the service instance just to make this more DRY.

**Review by marcelosalloum — APPROVED** [review]
LGTM!

The test is failing probably because the dist account is out of funds. That account's public key should not be a secret, it makes it hard for us to debug it right now. @marwen-abid said he'd take a look.

Once we make that test pass, we can emerge this PR, is that ok?

**Pull request merged**